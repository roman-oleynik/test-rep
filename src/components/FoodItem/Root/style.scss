@import '../../../utils/scss/utils.scss';
@import '../../../assets/fonts/css/fonts.css';

.food-item {
    @include flex(column, flex-start, center, nowrap);

    .food-item__label {
        .food-item__checkbox {
            @include setSize(1px, 1px);
            opacity: 0;
        }
        .food-item__checkbox:focus + .food-item__content {
            border: 0.15vw solid $white;
        }
        .food-item__checkbox:checked + .food-item__content > .button-shape {
            stroke: $violet;
        }
        .food-item__checkbox:disabled + .food-item__content > .button-shape {
            stroke: $grey;
        }
        .food-item__checkbox:checked:disabled + .food-item__content > .button-shape {
            stroke: $grey;
        }
        .food-item__content {
            position: relative;
            background: url(../../../svg.svg);
            border: 0.15vw solid transparent;
            outline: none;
            @extend .font;
            overflow: hidden;

            .button-shape {
                position: relative;
                width: 25vw;
                fill: $lightgrey;
                stroke: $blue;
                stroke-width: 6;
            }
        }

        /* For EDGE */
        .food-item__content_dis .button-shape {
            stroke: $grey;
        }
    }
}

@media (max-width: 992px) {
    .food-item {
        @include flex(column, flex-start, center, wrap);
        flex-basis: 270px;

        .food-item__label {
            .food-item__checkbox:focus + .food-item__content {
                border: 2px solid $white;
            }
            .food-item__content {
                border: 2px solid transparent;
    
                .button-shape {
                    position: relative;
                    @include setSize(370px, 250px);
                }
            }
        }
    }
}