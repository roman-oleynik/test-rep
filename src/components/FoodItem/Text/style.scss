@import '../../../utils/scss/utils.scss';


.food-item__header {
    position: absolute;
    top: 0;
    left: 2vw;
    @include flex(column, flex-start, flex-start, nowrap);
    padding-left: 3vw;
    
    .food-item__dislike {
        color: $violet;
        background: transparent;
        border: 0;
        padding: 1vw 0;
        @extend .font;
        font-size: 1.35vw;
    }
    .food-item__text {
        font-size: 1.35vw;
        padding: 1vw 0;
    }
    .food-item__title {
        font-size: 3.4vw;
        line-height: 3.4vw;
        text-align: left;

        .small-title {
            font-size: 2.5vw;
        }
    }
    .food-item__list {
        padding-top: 1vw;
        list-style: none;
        @include flex(column, flex-start, flex-start);

        .food-item__item {
            font-size: $small;
        }
    }
}
.food-item__header_dis {
    opacity: 0.5;
}

@media (max-width: 992px) {
    .food-item__header {
        left: 20px;
        padding-left: 30px;
        
        .food-item__dislike {
            padding: 10px 0;
            font-size: 14px;
        }
        .food-item__text {
            font-size: 14px;
            padding: 10px 0;
        }
        .food-item__title {
            font-size: 32px;
            line-height: 32px;
    
            .small-title {
                font-size: 24px;
            }
        }
        .food-item__list {
            padding-top: 10px;
    
            .food-item__item {
                font-size: 14px;
            }
        }
    }
}