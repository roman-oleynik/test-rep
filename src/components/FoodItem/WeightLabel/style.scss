@import '../../../utils/scss/utils.scss';

.food-item__circle {
    position: absolute;
    bottom: 1.8vw;
    right: 1.8vw;
    @include flex(column, center, center, nowrap);
    @include setSize(6.3vw, 6.3vw);
    @extend .round-off;
    background: $blue;

    .food-item__weight {
        color: $white;
        font-size: 3vw;
        line-height: 3vw;
    }
    
    .font-smaller {
        font-size: 2vw;
        line-height: 2vw;
        color: $white;
    }
    
}
.food-item__circle_sel {
    background: $violet;
}
.food-item__circle_dis {
    background: $grey;
}
.food-item__circle_sel-over {
    background: $lightviolet;
}
.food-item__circle_over {
    background: $lightblue;
}

@media (max-width: 992px) {
    .food-item__circle {
        bottom: 18px;
        right: 18px;
        @include setSize(62px, 62px);
    
        .food-item__weight {
            font-size: 30px;
            line-height: 30px;
        }
        
        .font-smaller {
            font-size: 20px;
            line-height: 20px;
        }
        
    }
}