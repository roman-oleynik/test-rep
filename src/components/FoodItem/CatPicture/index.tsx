import React from 'react';
import cat from '../../../assets/img/cat.png';

import './style.scss';

type Props = {
    isDisabled: boolean
};

export function CatPicture({isDisabled}: Props) {
    return <div className="food-item__mask">
        <img 
            src={cat} 
            className={`food-item__image ${isDisabled ? "food-item__image_dis" : ""}`}
            alt="Кот смотрит прямо в душу."
        />
    </div>
}