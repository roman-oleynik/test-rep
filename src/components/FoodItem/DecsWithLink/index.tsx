import React from 'react';
import './style.scss';

export function DecsWithLink() {
    return <>
        {"Чего стоишь? Порадуй котэ, "}
        <a aria-label="Купить корм" href="#" className="reset-link-style link_blue">купи.</a>
    </>
}