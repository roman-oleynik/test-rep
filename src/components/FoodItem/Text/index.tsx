import React from 'react';
import { mousesToStr } from '../../../utils/ts/mousesToStr';

import './style.scss';
import { CatFoodItem } from '../../../model';


type Props = {
    isMouseOver: boolean
    isSelected: boolean
    isDisabled: boolean
    model: CatFoodItem
};

export function TextOfCard({
        isMouseOver, 
        isSelected, 
        isDisabled, 
        model
    }: Props) {
    const { filling, portions, mouses, isClientSatisfied } = model;

    return <header className={`food-item__header ${isDisabled ? "food-item__header_dis" : ""}`}>
        {
            !isMouseOver && isSelected && !isDisabled
            ?
            <p className="food-item__dislike">Котэ не одобряет?</p>
            :
            <p className="food-item__text">Сказочное заморское яство</p>
        }
        <h3 className="food-item__title">
            Нямушка <br/>
            <span className="small-title">{filling}</span>
        </h3>
        <ul className="food-item__list">
            <li className="food-item__item">{portions} порций</li>
            <li className="food-item__item">{`${mousesToStr(mouses)} в подарок`}</li>
            {
                isClientSatisfied && <li className="food-item__item">заказчик доволен</li>
            }
        </ul>
    </header>
}