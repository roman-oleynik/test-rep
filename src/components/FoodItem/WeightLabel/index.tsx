import React from 'react';
import './style.scss';

type Props = {
    isMouseOver: boolean
    isSelected: boolean
    isDisabled: boolean
    weight: number
};

export function WeightLabel({isMouseOver, isSelected, isDisabled, weight}: Props) {
    const weightClassList: string = `
        food-item__circle 
        ${isMouseOver && isSelected && !isDisabled ? "food-item__circle_sel-over" : ""}
        ${isMouseOver && !isSelected && !isDisabled ? "food-item__circle_over" : ""} 
        ${isSelected && !isDisabled ? "food-item__circle_sel" : ""}
        ${isDisabled ? "food-item__circle_dis" : ""}
    `;

    return <span className={weightClassList}>
        <p className="food-item__weight aligner">
            {weight}
        </p>
        <span className="font-smaller">кг</span>
    </span>
}