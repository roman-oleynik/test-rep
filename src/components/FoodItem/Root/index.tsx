import React, { useState } from 'react';

import './style.scss';
import { DecsWithLink } from '../DecsWithLink';
import { WeightLabel } from '../WeightLabel';
import { CatPicture } from '../CatPicture';
import { TextOfCard } from '../Text';
import { CatFoodItem } from '../../../model';


type Props = {
    model: CatFoodItem
};


export function FoodItem({ model }: Props) {
    const { weight, subContent, onDisable, filling } = model;

    const [isSelected, setIsSelected] = useState(false);
    const [isDisabled, setIsDisabled] = useState(false);
    const [isMouseOver, setIsMouseOver] = useState(false);

    const onCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setIsSelected(e.target.checked);
    };

    const disable = () => {
        setIsDisabled(true);
    }

    return (
        <section className="food-item" aria-label={`Нямушка ${filling}`}>
            <label className="food-item__label">
                <input
                    aria-hidden="true"
                    className="food-item__checkbox"
                    type="checkbox"
                    disabled={isDisabled}
                    onChange={onCheckboxChange}
                />
                <div
                    className={`food-item__content ${isDisabled ? "food-item__content_dis" : ""}`}
                    onMouseOver={() => setIsMouseOver(true)}
                    onMouseOut={() => setIsMouseOver(false)}
                >
                    <svg className="button-shape" x="0px" y="0px" viewBox="0 0 328.5 488.5">
                        <path d="M63.7,4.3h250.6c5.5,0,10,4.5,10,10v460c0,5.5-4.5,10-10,10h-300c-5.5,0-10-4.5-10-10V59.9L63.7,4.3z"/>
                    </svg>
                    <TextOfCard
                        isMouseOver = {isMouseOver}
                        isSelected = {isSelected}
                        isDisabled = {isDisabled}
                        disable = {disable}
                        model = {model}
                    />
                    <CatPicture isDisabled = {isDisabled} />
                    <WeightLabel
                        isMouseOver = {isMouseOver}
                        isSelected = {isSelected}
                        isDisabled = {isDisabled}
                        weight = {weight}
                    />
                </div>
            </label>
            
            <p className="food-item__description">
                {isDisabled && onDisable}
                {isSelected && !isDisabled && subContent}
                {!isSelected && !isDisabled && 
                    <DecsWithLink />
                }
            </p>
        </section>
    );
}