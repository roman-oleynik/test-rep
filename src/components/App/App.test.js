import React from 'react';
import renderer from 'react-test-renderer';
import { App } from './index';
import { FoodItem } from '../FoodItem/Root';
import { TextOfCard } from '../FoodItem/Text';
import { WeightLabel } from '../FoodItem/WeightLabel';
import { model } from '../../model';


describe("App.js integration tests", () => {
    it("should change state on the checkbox change", () => {
        const component = renderer.create(<App model={model} />);
        expect(component.toJSON()).toMatchSnapshot();

        const foodItem = renderer.create(<FoodItem model={model[0]} />);
        const checkbox = foodItem.root.findByProps({className: "food-item__checkbox"});
        
        checkbox.props.onChange({target: {checked: true}});
        expect(foodItem.toJSON()).toMatchSnapshot();

        checkbox.props.onChange({target: {checked: false}});
        expect(foodItem.toJSON()).toMatchSnapshot();

        checkbox.props.onChange({target: {checked: true}});
        expect(foodItem.toJSON()).toMatchSnapshot();
    });
    it("sholud highlight the 'WeightLabel' after a cursor hover", () => {
        const foodItem = renderer.create(<FoodItem model={model[0]} />);
        const container = foodItem.root.findByProps({className: "food-item__content "});
        container.props.onMouseOver();
        expect(foodItem.root.findByType(WeightLabel).props.isMouseOver).toBe(true);
    });
    it("should show the 'Котэ не одобряет' after a click and a mouseover", () => {
        const disable = jest.fn();
        const component = renderer.create(
            <TextOfCard
                isMouseOver = {true}
                isSelected = {true}
                isDisabled = {false}
                disable = {disable}
                model = {model[0]}
            />
        );
        expect(component.root.findByProps({className: "food-item__dislike"}).props.children).toBe("Котэ не одобряет?");
    });
    it("should call the 'disable' function once after the click the 'Котэ не одобряет?' button", () => {
        const disable = jest.fn().mockImplementation(() => {});
        const component = renderer.create(
            <TextOfCard
                isMouseOver = {true}
                isSelected = {true}
                isDisabled = {false}
                disable = {disable}
                model = {model[0]}
            />
        );
        const dislike = component.root.findByProps({className: "food-item__dislike"});
        dislike.props.onClick();
        expect(disable).toHaveBeenCalledTimes(1);
    });
})