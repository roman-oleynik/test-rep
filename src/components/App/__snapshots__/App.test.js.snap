// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js integration tests should change state on the checkbox change 1`] = `
<div
  className="app"
>
  <header />
  <main
    className="main"
  >
    <article
      className="cat-food"
    >
      <header>
        <h2
          className="cat-food__title"
        >
          Ты сегодня покормил кота?
        </h2>
      </header>
      <section
        className="cat-food-content"
      >
        <section
          aria-label="Нямушка с фуа-гра"
          className="food-item"
        >
          <label
            className="food-item__label"
          >
            <input
              aria-hidden="true"
              className="food-item__checkbox"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            <div
              className="food-item__content "
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <svg
                className="button-shape"
                viewBox="0 0 328.5 488.5"
                x="0px"
                y="0px"
              >
                <path
                  d="M63.7,4.3h250.6c5.5,0,10,4.5,10,10v460c0,5.5-4.5,10-10,10h-300c-5.5,0-10-4.5-10-10V59.9L63.7,4.3z"
                />
              </svg>
              <header
                className="food-item__header "
              >
                <p
                  className="food-item__text"
                >
                  Сказочное заморское яство
                </p>
                <h3
                  className="food-item__title"
                >
                  Нямушка 
                  <br />
                  <span
                    className="small-title"
                  >
                    с фуа-гра
                  </span>
                </h3>
                <ul
                  className="food-item__list"
                >
                  <li
                    className="food-item__item"
                  >
                    10
                     порций
                  </li>
                  <li
                    className="food-item__item"
                  >
                    мышь в подарок
                  </li>
                </ul>
              </header>
              <div
                className="food-item__mask"
              >
                <img
                  alt="Кот смотрит прямо в душу."
                  className="food-item__image "
                  src="cat.png"
                />
              </div>
              <span
                className="
        food-item__circle 
        
         
        
        
    "
              >
                <p
                  className="food-item__weight aligner"
                >
                  0.5
                </p>
                <span
                  className="font-smaller"
                >
                  кг
                </span>
              </span>
            </div>
          </label>
          <p
            className="food-item__description"
          >
            Чего стоишь? Порадуй котэ, 
            <a
              aria-label="Купить корм"
              className="reset-link-style link_blue"
              href="#"
            >
              купи.
            </a>
          </p>
        </section>
        <section
          aria-label="Нямушка с рыбой"
          className="food-item"
        >
          <label
            className="food-item__label"
          >
            <input
              aria-hidden="true"
              className="food-item__checkbox"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            <div
              className="food-item__content "
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <svg
                className="button-shape"
                viewBox="0 0 328.5 488.5"
                x="0px"
                y="0px"
              >
                <path
                  d="M63.7,4.3h250.6c5.5,0,10,4.5,10,10v460c0,5.5-4.5,10-10,10h-300c-5.5,0-10-4.5-10-10V59.9L63.7,4.3z"
                />
              </svg>
              <header
                className="food-item__header "
              >
                <p
                  className="food-item__text"
                >
                  Сказочное заморское яство
                </p>
                <h3
                  className="food-item__title"
                >
                  Нямушка 
                  <br />
                  <span
                    className="small-title"
                  >
                    с рыбой
                  </span>
                </h3>
                <ul
                  className="food-item__list"
                >
                  <li
                    className="food-item__item"
                  >
                    40
                     порций
                  </li>
                  <li
                    className="food-item__item"
                  >
                    2 мыши в подарок
                  </li>
                </ul>
              </header>
              <div
                className="food-item__mask"
              >
                <img
                  alt="Кот смотрит прямо в душу."
                  className="food-item__image "
                  src="cat.png"
                />
              </div>
              <span
                className="
        food-item__circle 
        
         
        
        
    "
              >
                <p
                  className="food-item__weight aligner"
                >
                  2
                </p>
                <span
                  className="font-smaller"
                >
                  кг
                </span>
              </span>
            </div>
          </label>
          <p
            className="food-item__description"
          >
            Чего стоишь? Порадуй котэ, 
            <a
              aria-label="Купить корм"
              className="reset-link-style link_blue"
              href="#"
            >
              купи.
            </a>
          </p>
        </section>
        <section
          aria-label="Нямушка с курой"
          className="food-item"
        >
          <label
            className="food-item__label"
          >
            <input
              aria-hidden="true"
              className="food-item__checkbox"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            <div
              className="food-item__content "
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <svg
                className="button-shape"
                viewBox="0 0 328.5 488.5"
                x="0px"
                y="0px"
              >
                <path
                  d="M63.7,4.3h250.6c5.5,0,10,4.5,10,10v460c0,5.5-4.5,10-10,10h-300c-5.5,0-10-4.5-10-10V59.9L63.7,4.3z"
                />
              </svg>
              <header
                className="food-item__header "
              >
                <p
                  className="food-item__text"
                >
                  Сказочное заморское яство
                </p>
                <h3
                  className="food-item__title"
                >
                  Нямушка 
                  <br />
                  <span
                    className="small-title"
                  >
                    с курой
                  </span>
                </h3>
                <ul
                  className="food-item__list"
                >
                  <li
                    className="food-item__item"
                  >
                    100
                     порций
                  </li>
                  <li
                    className="food-item__item"
                  >
                    5 мышей в подарок
                  </li>
                  <li
                    className="food-item__item"
                  >
                    заказчик доволен
                  </li>
                </ul>
              </header>
              <div
                className="food-item__mask"
              >
                <img
                  alt="Кот смотрит прямо в душу."
                  className="food-item__image "
                  src="cat.png"
                />
              </div>
              <span
                className="
        food-item__circle 
        
         
        
        
    "
              >
                <p
                  className="food-item__weight aligner"
                >
                  5
                </p>
                <span
                  className="font-smaller"
                >
                  кг
                </span>
              </span>
            </div>
          </label>
          <p
            className="food-item__description"
          >
            Чего стоишь? Порадуй котэ, 
            <a
              aria-label="Купить корм"
              className="reset-link-style link_blue"
              href="#"
            >
              купи.
            </a>
          </p>
        </section>
      </section>
    </article>
  </main>
  <footer />
</div>
`;

exports[`App.js integration tests should change state on the checkbox change 2`] = `
<section
  aria-label="Нямушка с фуа-гра"
  className="food-item"
>
  <label
    className="food-item__label"
  >
    <input
      aria-hidden="true"
      className="food-item__checkbox"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="food-item__content "
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <svg
        className="button-shape"
        viewBox="0 0 328.5 488.5"
        x="0px"
        y="0px"
      >
        <path
          d="M63.7,4.3h250.6c5.5,0,10,4.5,10,10v460c0,5.5-4.5,10-10,10h-300c-5.5,0-10-4.5-10-10V59.9L63.7,4.3z"
        />
      </svg>
      <header
        className="food-item__header "
      >
        <p
          className="food-item__text"
        >
          Сказочное заморское яство
        </p>
        <h3
          className="food-item__title"
        >
          Нямушка 
          <br />
          <span
            className="small-title"
          >
            с фуа-гра
          </span>
        </h3>
        <ul
          className="food-item__list"
        >
          <li
            className="food-item__item"
          >
            10
             порций
          </li>
          <li
            className="food-item__item"
          >
            мышь в подарок
          </li>
        </ul>
      </header>
      <div
        className="food-item__mask"
      >
        <img
          alt="Кот смотрит прямо в душу."
          className="food-item__image "
          src="cat.png"
        />
      </div>
      <span
        className="
        food-item__circle 
        
         
        food-item__circle_sel
        
    "
      >
        <p
          className="food-item__weight aligner"
        >
          0.5
        </p>
        <span
          className="font-smaller"
        >
          кг
        </span>
      </span>
    </div>
  </label>
  <p
    className="food-item__description"
  >
    Печень утки разварная с артишоками.
  </p>
</section>
`;

exports[`App.js integration tests should change state on the checkbox change 3`] = `
<section
  aria-label="Нямушка с фуа-гра"
  className="food-item"
>
  <label
    className="food-item__label"
  >
    <input
      aria-hidden="true"
      className="food-item__checkbox"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="food-item__content "
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <svg
        className="button-shape"
        viewBox="0 0 328.5 488.5"
        x="0px"
        y="0px"
      >
        <path
          d="M63.7,4.3h250.6c5.5,0,10,4.5,10,10v460c0,5.5-4.5,10-10,10h-300c-5.5,0-10-4.5-10-10V59.9L63.7,4.3z"
        />
      </svg>
      <header
        className="food-item__header "
      >
        <p
          className="food-item__text"
        >
          Сказочное заморское яство
        </p>
        <h3
          className="food-item__title"
        >
          Нямушка 
          <br />
          <span
            className="small-title"
          >
            с фуа-гра
          </span>
        </h3>
        <ul
          className="food-item__list"
        >
          <li
            className="food-item__item"
          >
            10
             порций
          </li>
          <li
            className="food-item__item"
          >
            мышь в подарок
          </li>
        </ul>
      </header>
      <div
        className="food-item__mask"
      >
        <img
          alt="Кот смотрит прямо в душу."
          className="food-item__image "
          src="cat.png"
        />
      </div>
      <span
        className="
        food-item__circle 
        
         
        
        
    "
      >
        <p
          className="food-item__weight aligner"
        >
          0.5
        </p>
        <span
          className="font-smaller"
        >
          кг
        </span>
      </span>
    </div>
  </label>
  <p
    className="food-item__description"
  >
    Чего стоишь? Порадуй котэ, 
    <a
      aria-label="Купить корм"
      className="reset-link-style link_blue"
      href="#"
    >
      купи.
    </a>
  </p>
</section>
`;

exports[`App.js integration tests should change state on the checkbox change 4`] = `
<section
  aria-label="Нямушка с фуа-гра"
  className="food-item"
>
  <label
    className="food-item__label"
  >
    <input
      aria-hidden="true"
      className="food-item__checkbox"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="food-item__content "
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <svg
        className="button-shape"
        viewBox="0 0 328.5 488.5"
        x="0px"
        y="0px"
      >
        <path
          d="M63.7,4.3h250.6c5.5,0,10,4.5,10,10v460c0,5.5-4.5,10-10,10h-300c-5.5,0-10-4.5-10-10V59.9L63.7,4.3z"
        />
      </svg>
      <header
        className="food-item__header "
      >
        <p
          className="food-item__text"
        >
          Сказочное заморское яство
        </p>
        <h3
          className="food-item__title"
        >
          Нямушка 
          <br />
          <span
            className="small-title"
          >
            с фуа-гра
          </span>
        </h3>
        <ul
          className="food-item__list"
        >
          <li
            className="food-item__item"
          >
            10
             порций
          </li>
          <li
            className="food-item__item"
          >
            мышь в подарок
          </li>
        </ul>
      </header>
      <div
        className="food-item__mask"
      >
        <img
          alt="Кот смотрит прямо в душу."
          className="food-item__image "
          src="cat.png"
        />
      </div>
      <span
        className="
        food-item__circle 
        
         
        food-item__circle_sel
        
    "
      >
        <p
          className="food-item__weight aligner"
        >
          0.5
        </p>
        <span
          className="font-smaller"
        >
          кг
        </span>
      </span>
    </div>
  </label>
  <p
    className="food-item__description"
  >
    Печень утки разварная с артишоками.
  </p>
</section>
`;
